source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/imu/imu.py

```diff
@@ -0,0 +1,197 @@
+from .imu_data import IMUData
+
+if TYPE_CHECKING:
+    from .imu_cfg import IMUCfg
+
+
+class IMU(SensorBase):
+    """The inertia measurement unit."""
+
+    cfg: IMUCfg
+    """The configuration parameters."""
+
+    def __init__(self, cfg: IMUCfg):
+        """Initializes the IMU sensor.
+
+        Args:
+            cfg: The configuration parameters.
+        """
+        # initialize base class
+        super().__init__(cfg)
+        # Create empty variables for storing output data
+        self._data = IMUData()
+
+    def __str__(self) -> str:
+        """Returns: A string containing information about the instance."""
+        return (
+            f"IMU sensor @ '{self.cfg.prim_path}': \n"
+            f"\tview type         : {self._view.__class__}\n"
+            f"\tupdate period (s) : {self.cfg.update_period}\n"
+            f"\tnumber of sensors : {self._view.count}\n"
+        )
+
+    """
+    Properties
+    """
+
+    @property
+    def data(self) -> IMUData:
+        # update sensors if needed
+        self._update_outdated_buffers()
+        # return the data
+        return self._data
+
+    @property
+    def num_instances(self) -> int:
+        return self._view.count
+
+    """
+    Operations
+    """
+
+    def reset(self, env_ids: Sequence[int] | None = None):
+        # reset the timestamps
+        super().reset(env_ids)
+        # resolve None
+        if env_ids is None:
+            env_ids = slice(None)
+        # reset accumulative data buffers
+        self._data.quat_w[env_ids] = 0.0
+        self._data.ang_vel_b[env_ids] = 0.0
+        self._data.lin_acc_b[env_ids] = 0.0
+
+    def update(self, dt: float, force_recompute: bool = False):
+        # save timestamp
+        self._dt = dt
+        # execute updating
+        super().update(dt, force_recompute)
+
+    """
+    Implementation.
+    """
+
+    def _initialize_impl(self):
+        """Initializes the sensor handles and internal buffers.
+
+        This function creates handles and registers the provided data types with the replicator registry to
+        be able to access the data from the sensor. It also initializes the internal buffers to store the data.
+
+        Raises:
+            RuntimeError: If the number of imu prims in the view does not match the number of environments.
+            RuntimeError: If the imu prim is not a RigidBodyPrim
+        """
+        # Initialize parent class
+        super()._initialize_impl()
+        # create simulation view
+        self._physics_sim_view = physx.create_simulation_view(self._backend)
+        self._physics_sim_view.set_subspace_roots("/")
+        # check if the prim at path is a rigid prim
+        prim = sim_utils.find_first_matching_prim(self.cfg.prim_path)
+        if prim is None:
+            raise RuntimeError(f"Failed to find a prim at path expression: {self.cfg.prim_path}")
+        # check if it is a RigidBody Prim
+        if prim.HasAPI(UsdPhysics.RigidBodyAPI):
```

Comment:
Above you mentioned the prim can be articulation too but that isn't supported, right?